src/neural/cuda/network_cudnn.cc

```diff
@@ -950,7 +954,7 @@ class CudnnNetwork : public Network {
   // Apparently nvcc doesn't see constructor invocations through make_unique.
   // This function invokes constructor just to please complier and silence
   // warning. Is never called (but compiler thinks that it could).
-  void UglyFunctionToSilenceNvccWarning() { InputsOutputs io(0, false, false); }
+  void UglyFunctionToSilenceNvccWarning() { InputsOutputs io(0, false, false, false); }
 
  private:
   const NetworkCapabilities capabilities_;
```

Comment:
Later in the file, after line 1098: `case pblczero::NetworkFormat::NETWORK_ATTENTIONBODY_WITH_HEADFORMAT:` you should add the `case pblczero::NetworkFormat::NETWORK_ATTENTIONBODY_WITH_MULTIHEADFORMAT:` to switch to cuda.

Reply:
Done